### ISSUE TITLE
mkdir: cannot create directory '/opt/bitnami/nginx/conf/bitnami': Read-only file system

### ISSUE THREAD

**Issue by jgournet** [bitnami/charts#16391]
### Name and Version

bitnami/nginx

### What architecture are you using?

amd64

### What steps will reproduce the bug?

values.yaml file:
```yaml
extraDeploy:
  - |
    apiVersion: v1
    kind: ConfigMap
    metadata:
      name: nginx-test-conf
      namespace: kube-system
    data:
      nginx.conf: |
        server {
          my config here - this is NOT a valid config, I know :)
        }


extraVolumeMounts:
  - name: nginx-conf
    mountPath: /opt/bitnami/nginx/conf/
  - mountPath: /tmp
    name: tmp-volume

extraVolumes:
  - name: nginx-conf
    configMap:
      name: nginx-test-conf
      items:
        - key: nginx.conf
          path: nginx.conf
  - name: tmp-volume
    emptyDir: {}
```

Installing helm chart with the above config:
```
helm install my-release -f /tmp/values.yaml -n kube-system oci://registry-1.docker.io/bitnamicharts/nginx
```
=> we get the error from title

### What is the expected behavior?

Same behavior as this:
```
helm install my-release -f /tmp/values.yaml --set image.tag=1.23.3-debian-11-r26 -n kube-system oci://registry-1.docker.io/bitnamicharts/nginx
```
ie: things work
(ok, in this case, my nginx config is wrong, so we get a proper "server directive is not allowed here" :) )

### What do you see instead?

```
mkdir: cannot create directory '/opt/bitnami/nginx/conf/bitnami': Read-only file system
```

### Additional information

You might ask "why are you doing this ??"
This is because we want to add "stream" blocks, so we write our own full config.

It was working well, until this was introduced:
File: /opt/bitnami/scripts/nginx/setup.sh
```
# Regenerate SSL certs (without a passphrase)
ensure_dir_exists "${NGINX_CONF_DIR}/bitnami/certs"
if [[ ! -f "${NGINX_CONF_DIR}/bitnami/certs/server.crt" ]]; then
    SSL_KEY_FILE="${NGINX_CONF_DIR}/bitnami/certs/server.key"
[...]
```
this "ensure_dir_exists" fails in our case, as `NGINX_CONF_DIR` is indeed a read-only FS as it comes from the config map.

=>
Could you see if you can do a fix for this?
(or maybe let me know how I can add stream server blocks a bit more properly than overwriting the whole config)

**Comment by jgournet** [comment]
Update:
1) Easy fix idea: What about adding a test before creating those certs ? like adding an ENV variable "DO_NOT_CREATE_CERTS". If set: don't create certs :) That should be backward compatible, and that would help us enough

2) really not proud of this workaround, but it should get us going (ie: allow us to use "latest" image) until you can help us with a better solution (or until you do an improvement on those certs that will break it for us)
```
extraDeploy:
  - |
    apiVersion: v1
    kind: ConfigMap
    metadata:
      name: nginx-test-conf
      namespace: kube-system
    data:
      server.crt: |
        nothing to see here - this file is not set to be read by nginx
      nginx.conf: |
        server {
          my config here - this is NOT a valid config, I know :)
        }


extraVolumeMounts:
  - name: nginx-conf
    mountPath: /opt/bitnami/nginx/conf/
  - name: dummy-cert
    mountPath: /opt/bitnami/nginx/conf/bitnami/certs
  - mountPath: /tmp
    name: tmp-volume

extraVolumes:
  - name: nginx-conf
    configMap:
      name: nginx-test-conf
      items:
        - key: nginx.conf
          path: nginx.conf
  - name: dummy-cert  # workaround for https://github.com/bitnami/charts/issues/16391
    configMap:
      name: {{ include "common.names.fullname" . }}-conf
      items:
        - key: server.crt
          path: server.crt
  - name: tmp-volume
    emptyDir: {}
```

**Comment by migruiz4** [comment]
Hi @jgournet,

I'm sorry for the late response.

I think this case may be related to this issue: https://github.com/bitnami/containers/issues/30774

We already added a fix for the read-only filesystem issue caused by the certificate generation logic.
The fix is available in the images `bitnami/nginx:1.24.0-debian-11-r8` and later, and the latest version of the chart (`14.2.2`).

Could you please try upgrading the chart and let us know if the issue persists?

**Comment by jgournet** [comment]
@migruiz4 
Thanks for finding the time to reply to this ticket - I should have opened it in "containers" ;)
Anyway: I can confirm it's working great !
